sync/src/block_status.rs

```diff
@@ -0,0 +1,81 @@
+use bitflags::bitflags;
+
+bitflags! {
+    pub struct BlockStatus: u32 {
+        const UNKNOWN                 =     0;
+
+        const HEADER_VERIFIED         =     1;
+        const BLOCK_RECEIVED          =     Self::HEADER_VERIFIED.bits | 2;
+        const BLOCK_STORED            =     Self::HEADER_VERIFIED.bits | Self::BLOCK_RECEIVED.bits | 16;
```

Comment:
Should clear old stored block in case of fork.

Reply:
Discussed.